ff/src/biginteger/tests.rs

```diff
@@ -49,6 +50,33 @@ fn biginteger_arithmetic_test<B: BigInteger>(a: B, b: B, zero: B) {
     let mut a_plus_a = a;
     a_plus_a.add_with_carry(&a); // Won't assert anything about carry bit.
     assert_eq!(a_mul2, a_plus_a);
+
+    // a * 1 = a
```

Comment:
Let's also add some tests to `test-templates`, if appropriate?

Reply:
For these implementations around `BigInt`, all unit tests can be found in this file:
- So I would say that for the moment there is not necessarily a need to outsource the tests to `test-templates` no?
- I added a `max` parameter to try to test with a slightly wider coverage of all the multiplication features
- If you want I can also add another function here that tests something specifically for multiplication? But it seems to me that with the tests here we are pretty well covered.
- To avoid mixing the PRs, if you want we can add a new issue to migrate all the tests that are here to `test-templates` and try to put an even broader conversion on all the functions of the implementation, I don't know if you see any interest in it?